PayPalWebPayments/src/main/java/com/paypal/android/paypalwebpayments/PayPalWebCheckoutClient.kt

```diff
@@ -96,11 +127,28 @@ class PayPalWebCheckoutClient internal constructor(
 
     internal fun handleBrowserSwitchResult() {
         browserSwitchResult = browserSwitchClient.deliverResult(activity)
-        listener?.also {
-            browserSwitchResult?.also { result ->
-                when (result.status) {
-                    BrowserSwitchStatus.SUCCESS -> deliverSuccess()
-                    BrowserSwitchStatus.CANCELED -> deliverCancellation()
+        if (listener == null && vaultListener == null) {
```

Comment:
Preserving functionality here, but the original logic may need to be re-visited. We should also move to [manual browser switching](https://github.com/braintree/braintree_android/blob/main/v4.9.0%2B_MIGRATION_GUIDE.md#manual-browser-switching-for-browser-based-flows) in a next major version to align with `braintree_android` and prevent logic like this from being necessary.